flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/AbstractHeapState.java

```diff
@@ -42,33 +42,35 @@
 	/** Map containing the actual key/value pairs. */
 	protected final StateTable<K, N, SV> stateTable;
 
-	/** This holds the name of the state and can create an initial default value for the state. */
-	protected final SD stateDesc;
```

Comment:
I think with removing this, you can also remove SD from the generic type of this class, and transitively from the subclasses as well.